src/miniscript/satisfy.rs

```diff
@@ -94,7 +95,13 @@ pub struct Older(pub u32);
 
 impl<Pk: MiniscriptKey> Satisfier<Pk> for Older {
     fn check_older(&self, n: u32) -> bool {
-        n <= self.0
+        if !((n < HEIGHT_TIME_THRESHOLD && self.0 < HEIGHT_TIME_THRESHOLD)
```

Comment:
It may be simpler to check either
```
(n < HEIGHT_TIME_THRESHOLD) == (self.0 < HEIGHT_TIME_THRESHOLD)
```
or just
```
n < HEIGHT_TIME_THRESHOLD && self.0 >= HEIGHT_TIME_THRESHOLD
```

Reply:
My logic for these is trying to implement _as it is_ written in the BIP.

Reply:
Which BIP?

My suggestion is logically equivalent to what's written here, for what it's worth, it's just shorter.

Reply:
https://github.com/bitcoin/bips/blob/master/bip-0065.mediawiki . I can change this too, just was a bit pedantic in trying to copy the logic conditions directly.